tests/Service/DatabaseManagerTest.php

```diff
@@ -140,6 +161,28 @@ public function testIsCreatingAndRollingBackTransaction(): void
         $this->assertTrue($db->rollBack());
     }
 
+    public function testIsCreatingAndRollingBackTransactionUsingExecuteTransaction(): void
+    {
+        $this->expectException(DatabaseException::class);
+
+        $db = new DatabaseManager();
+        $connectionOptions = [
+            'host' => TEST_DATABASE_HOST,
+            'port' => TEST_DATABASE_PORT,
+            'dbname' => TEST_DATABASE_DBNAME,
+            'username' => TEST_DATABASE_USERNAME,
+            'password' => TEST_DATABASE_PASSWORD,
+        ];
+        $db->addConnection(DatabaseManager::DRIVER_MYSQL, $connectionOptions, DatabaseManager::ROLE_MASTER);
+        $db->addConnection(DatabaseManager::DRIVER_MYSQL, $connectionOptions, DatabaseManager::ROLE_SLAVE);
+
+        $callable = function (): void {
+            throw new DatabaseException();
```

Comment:
Please execute an invalid statement after executing a valid one. Then test to make sure it was rolled back.